decksite: tooltip with the bug description for bugged cards

jgabrielygalan opened the pull request:
Fixes: #1633.

Commit:
decksite: tooltip with the bug description for bugged cards

bakert commented on decksite/static/js/pd.js:
```diff
@@ -162,6 +163,14 @@ PD.showLocalTimes = function () {
+        element.insertAfter($(this));
+        Tipped.create(element, $(this).attr("data-bug-desc"), {maxWidth: "200"});
+    });
+}
```
Does doing it in js cause the bug icons to "blink" into the page after the page has already rendered? /bugs/ is a good place to check.

jgabrielygalan replied:
With the change of strategy of generating the div with the bug icon in mustache, this doesn't happen, so no problem here.

bakert commented on decksite/templates/entry.mustache:
```diff
@@ -1 +1 @@
-<li><a href="{{card.url}}" class="{{^card.pd_legal}}illegal{{/card.pd_legal}} {{#card.bug_desc}}bugged{{/card.bug_desc}} card">{{n}} {{name}}</a></li>
+<li><a href="{{card.url}}" class="{{^card.pd_legal}}illegal{{/card.pd_legal}} {{#card.bug_desc}}bugged{{/card.bug_desc}} card" {{#card.bug_desc}}data-bug-desc="{{card.bug_class}}: {{card.bug_desc}}"{{/card.bug_desc}}>{{n}} {{name}}</a></li>
```
We could write out the bug element with a title attribute here and the existing tooltip-adding code would add the tooltip and get rid of the need for initBugIcons?

jgabrielygalan replied:
I think you are right. I was obssessed with creating everything in js, I'll try to do as you say here.

bakert commented on decksite/static/css/pd.css:
```diff
@@ -859,9 +863,8 @@ p.bugged {
-    content: "🐞 ";
-    margin-left: 0.2em;
+.bugicon {
+    float:left;
```
If we don't clear anywhere does this cause cards with shorter names to bunch up more-than-one on a line?

Do we still need the margin-left to stop the bug icon touching the associated card? We want a small gap there.

jgabrielygalan replied:
There was already a clear:both in the .decklist that I thought would be enough, I'm not sure. I'll add the clear:both to the li too, just in case.

jgabrielygalan replied:
And you were right too with the margin-left :)

bakert reviewed (commented):
Cool stuff. Some suggestions but we can ship as-is if you're happy with it.

bakert committed:
Merge branch 'master' into bugged_cards_tooltip

bakert committed:
Merge branch 'master' into bugged_cards_tooltip

Commit:
decksite: fixes for the bug desc tooltip

Commit:
Merge remote-tracking branch 'origin/bugged_cards_tooltip' into bugged_cards_tooltip

jgabrielygalan committed:
Merge branch 'master' into bugged_cards_tooltip

bakert committed:
Two tweaks to bug desc tooltips.

    1. Use an inline span instead of a block level div so we don't need float.
       (Floating both elements took everything out-of-flow so line-height didn't
       apply and things become 'scrunched up' too close together).
    2. Make the bug icon a link to the card page. Something feels weird about
       getting a tool tip when your cursor is an 'I'.
